sdk/eventgrid/Azure.Messaging.EventGrid/src/MigrationGuide.md

```diff
@@ -0,0 +1,195 @@
+
+We assume that you are familiar with Microsoft.Azure.EventGrid. If not, please refer to the README for [Azure.Messaging.EventGrid](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/eventgrid/Azure.Messaging.EventGrid/README.md) rather than this guide.
+
+## Table of contents
+- [Migration benefits](#migration-benefits)
+  - [Cross Service SDK improvements](#cross-service-sdk-improvements)
+  - [New features](#new-features)
+- [Important changes](#important-changes)
+  - [Package names and namespaces](#package-names-and-namespaces)
+  - [Client naming and constructors](#client-naming-and-constructors)
+  - [Publishing events to a topic](#publishing-events-to-a-topic)
+  - [Deserializing events and data](#deserializing-events-and-data)
+
+## Migration benefits
+
+As Azure has matured and been embraced by a more diverse group of developers, we have been focused on learning the patterns and practices to best support developer productivity and to understand the gaps that the .NET client libraries have.
+
+There were several areas of consistent feedback expressed across the Azure client library ecosystem. One of the most important is that the client libraries for different Azure services have not had a consistent approach to organization, naming, and API structure. Additionally, many developers have felt that the learning curve was difficult, and the APIs did not offer a good, approachable, and consistent onboarding story for those learning Azure or exploring a specific Azure service.
+
+To improve the development experience across Azure services, a set of uniform [design guidelines](https://azure.github.io/azure-sdk/general_introduction.html) was created for all languages to drive a consistent experience with established API patterns for all services. A set of [.NET design guidelines](https://azure.github.io/azure-sdk/dotnet_introduction.html) was also introduced to ensure that .NET clients have a natural and idiomatic feel with respect to the .NET ecosystem. The new Azure.Messaging.EventGrid library follows these guidelines.
+
+### Cross Service SDK improvements
+
+The modern Azure.Messaging.EventGrid client library also provides the ability to share in some of the cross-service improvements made to the Azure development experience, such as
+
+- Using the new Azure.Identity library to share a single authentication approach between clients
+- A unified logging and diagnostics pipeline offering a common view of the activities across each of the client libraries
+
+### New features
+
+The new Azure.Messaging.EventGrid library includes several new features:
+- Publishing and deserializing [CloudEvents](https://cloudevents.io/) using the Azure.Core `CloudEvent` type
+- Methods that allow publishing a single event in addition to lists of events
+- Ability to use a custom serializer for the Data of an event
+- Ability to publish custom schema events
```

Comment:
```suggestion
- Ability to publish events with a custom schema
```

I'm not sure that I understand the context entirely,  I may be off-base.

Reply:
I think either wording is fine, to be honest.  The new feature here is exposing support for events which don't use the cloud events or event grid event schema, but rather [a user defined schema that they provide when they construct the topic](https://docs.microsoft.com/en-us/azure/event-grid/input-mappings).